src/test/java/graphql/annotations/GraphQLExtensionsTest.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations;
+
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLSchema;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static graphql.Scalars.GraphQLString;
+import static graphql.schema.GraphQLSchema.newSchema;
+import static org.testng.Assert.assertEquals;
+
+public class GraphQLExtensionsTest {
+
+    @GraphQLDescription("TestObject object")
+    @GraphQLName("TestObject")
+    private static class TestObject {
+        @GraphQLField
+        public
+        String field() {
+            return "test";
+        }
+
+    }
+
+    @GraphQLTypeExtension(GraphQLExtensionsTest.TestObject.class)
+    private static class TestObjectExtension {
+        @GraphQLField
+        public static String field2() {
+            return "test2";
+        }
+    }
+
+    @Test
+    public void fields() {
+        GraphQLAnnotations.getInstance().registerTypeExtension(TestObjectExtension.class);
+        GraphQLObjectType object = GraphQLAnnotations.object(GraphQLExtensionsTest.TestObject.class);
+        GraphQLAnnotations.getInstance().unregisterTypeExtension(TestObjectExtension.class);
```

Comment:
Oh, I see now what you're trying to do. It makes sense thinking on an "extendable" model that can be shared and extended without modifying the original type. It would be good to have the opinion of some other more experienced contributor.
Will get back to you if I have further thoughts.